wgcfg/ip.go

```diff
@@ -0,0 +1,152 @@
+package wgcfg
+
+import (
+	"fmt"
+	"math"
+	"net"
+)
+
+// IP is an IPv4 or an IPv6 address.
+//
+// Internally the address is always represented in its IPv6 form.
+// IPv4 addresses use the IPv4-in-IPv6 syntax.
+type IP struct {
+	Addr [16]byte
+}
+
+func (ip IP) String() string { return net.IP(ip.Addr[:]).String() }
+
+// IP converts ip into a standard library net.IP.
+func (ip IP) IP() net.IP { return net.IP(ip.Addr[:]) }
+
+// Is6 reports whether ip is an IPv6 address.
+func (ip IP) Is6() bool { return !ip.Is4() }
+
+// Is4 reports whether ip is an IPv4 address.
+func (ip IP) Is4() bool {
+	return ip.Addr[0] == 0 && ip.Addr[1] == 0 &&
+		ip.Addr[2] == 0 && ip.Addr[3] == 0 &&
+		ip.Addr[4] == 0 && ip.Addr[5] == 0 &&
+		ip.Addr[6] == 0 && ip.Addr[7] == 0 &&
+		ip.Addr[8] == 0 && ip.Addr[9] == 0 &&
+		ip.Addr[10] == 0xff && ip.Addr[11] == 0xff
+}
+
+// To4 returns either a 4 byte slice for an IPv4 address, or nil if
+// it's not IPv4.
+func (ip IP) To4() []byte {
+	if ip.Is4() {
+		return ip.Addr[12:16]
+	} else {
+		return nil
+	}
+}
+
+// Equal reports whether ip == x.
+func (ip IP) Equal(x IP) bool {
+	return ip == x
+}
+
+func (ip IP) MarshalText() ([]byte, error) {
+	return []byte(ip.String()), nil
```

Comment:
Not super familiar here -- what's with the function name and signature? Can't we just have `String()` like usual? Is this for some weird reflection serialization?